### PR TITLE
Upload file module

### PR DESCRIPTION
**Pull request by kingiw**

### CODE REVIEW

File: web/src/DashboardPage.js [hunk 1]
```diff
@@ -1,14 +1,109 @@
...
 import {Link} from "react-router-dom";
 import Canvas from "./Canvas";
 import * as Backend from "./Backend";
 import TraceTable from "./TraceTable";
 
+
 const {Text} = Typography;
 
+
+class UploadFile extends React.Component {
```
Comment (hsluoyz) [inline web/src/DashboardPage.js]: Put it into `UploadFile.js`

File: web/src/DashboardPage.js [hunk 2]
```diff
@@ -1,14 +1,109 @@
...
 import * as Setting from "./Setting";
 import * as Shared from "./Shared";
-import {Table, Row, Col, Typography, Tag} from 'antd';
+import {Table, Row, Col, Typography, Tag, Icon, Button, Upload, message} from 'antd';
 import {Link} from "react-router-dom";
 import Canvas from "./Canvas";
 import * as Backend from "./Backend";
 import TraceTable from "./TraceTable";
 
+
```
Comment (hsluoyz) [inline web/src/DashboardPage.js]: No new line

File: controllers/test.go [hunk 3]
```diff
@@ -3,10 +3,12 @@ package controllers
 import (
 	"encoding/json"
 	"fmt"
+	"strconv"
 
 	"github.com/astaxie/beego"
 
 	"github.com/microsoft/mouselog/detect"
+	jsonutils "github.com/microsoft/mouselog/file"
```
Comment (hsluoyz) [inline controllers/test.go]: Why rename to `jsonutils`?

File: file/json_uils.go [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package jsonutils
```
Comment (hsluoyz) [inline file/json_uils.go]: file name typo

File: controllers/test.go [hunk 5]
```diff
@@ -90,3 +92,47 @@ func (c *ApiController) ClearTrace() {
 	c.Data["json"] = detect.GetDetectResult(ss, t.Id)
 	c.ServeJSON()
 }
+
+func (c *ApiController) UploadFile() {
```
Comment (hsluoyz) [inline controllers/test.go]: Put it into `dashboard.go`, as it's in Dashboard page.

File: controllers/test.go (hunk 3 above)
Comment (kingiw) [reply controllers/test.go]: 'file' is a very general identifier. Actually it is better to name the module "fileutils"

Comment (hsluoyz) [reply controllers/test.go]: OK.

**Comment by kingiw** [comment]
Commit updated. Plz review.

**Pull request merged**